donut/modules/marketplace/helpers.py

```diff
@@ -693,25 +656,397 @@ def process_edition(edition):
 
     try:
         edition = int(edition)
-        if edition < 1000:
-            # it's probably an edition, not a year
+        if edition < 100:
+            # It's probably an edition, not a year.
 
-            # if the tens digit is 1, it's always "th"
+            # If the tens digit is 1, it's always 'th'.
             if (edition / 10) % 10 == 1:
-                return str(edition) + "th"
+                return str(edition) + 'th'
             if edition % 10 == 1:
-                return str(edition) + "st"
+                return str(edition) + 'st'
             if edition % 10 == 2:
-                return str(edition) + "nd"
+                return str(edition) + 'nd'
             if edition % 10 == 3:
-                return str(edition) + "rd"
-            return str(edition) + "th"
+                return str(edition) + 'rd'
+            return str(edition) + 'th'
         else:
             return str(edition)
     except ValueError:
         return edition
 
 
+#################
+# TABLE QUERIES #
+#################
+def get_table_list_data(tables, fields=None, attrs={}):
+    """
+    Queries the database (specifically, table <table>) and returns list of
+    member data constrained by the specified attributes.
+
+    Arguments:
+        tables: The table(s) to query.  Ex: 'marketplace_items', or
+                'marketplace_items NATURAL LEFT JOIN
+                marketplace_textbooks'.
+        fields: The fields to return. If None specified, then default_fields
+                are used.
+        attrs:  The attributes of the members to filter for.
+    Returns:
+        result: The fields and corresponding values of members with desired
+                attributes. In the form of a list of lists.
+    """
+    if fields == None:
+        s_select_columns = 'SELECT * '
+    else:
+        s_select_columns = 'SELECT ' + ', '.join(fields) + ' '
+
+    s_from = 'FROM ' + tables
+
+    s_where = ''
+    if attrs:
+        s_where = ' WHERE '
+        s_where += ' AND '.join([key + ' = %s' for key in attrs])
+
+    s = s_select_columns + s_from + s_where
+
+    # Execute the query.
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(s, list(attrs.values()))
+        result = cursor.fetchall()
+
+    # Return the list of lists.
+    result = list(map(lambda a: list(a.values()), result))
```

Comment:
Hmm, this only works if fields isn't `None`; above, if fields is `None` I use a `SELECT *`.  I think I'll use both of your suggestions -- `return [list(row.values()) for row in result]` if fields is `None` and `return [[row[field] for field in fields] for row in result]` otherwise.

Reply:
Or, if `fields` was originally `None`, let `fields = list(result[0]) if result else []`